The below code is for the Shopify Data Science Summer Intern Tech Challenge. 

In [40]:
import pandas as pd
import numpy as np

In [33]:
data=pd.read_csv("/content/sample_data/2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv")

In [34]:
data.head()

order_id  shop_id  user_id  ...  total_items  payment_method           created_at
0         1       53      746  ...            2            cash  2017-03-13 12:36:56
1         2       92      925  ...            1            cash  2017-03-03 17:38:52
2         3       44      861  ...            1            cash   2017-03-14 4:23:56
3         4       18      935  ...            1     credit_card  2017-03-26 12:43:37
4         5       18      883  ...            1     credit_card   2017-03-01 4:35:11

[5 rows x 7 columns]

Q1. Think about what could be going wrong with our calculation. Think about a better way to evaluate this data.

In [36]:
total_revenue=data["order_amount"].sum()
total_revenue

15725640

In [37]:
AOV=total_revenue/len(data)
AOV

3145.128

There is actually nothing wrong with the Average Order Value calculation. By definition, Average Order Value is Total Revenue / Total Order = 15725640/5000 = 3145.128 USD. The value is high because there are a few very costly transactions in the middle, for example: for order_id 16, the order_amount is 704000, because of which the AOV is so high.However, the thing which is getting ignored is the total number of items per order as well. We need to consider the total_items per order while evaluating our data. If we need to evaluate the average price of a pair of sneakers, a better approach would be to evaluate the amount per item, and then total the same, and evaluate the average value over the number of orders. This would help us provide a more relatively accurate value per sneaker per order. For instance, for order_id 16, the cost per sneakers’ pair is 704000/2000 = $ 352, which makes sense. Another way to solve this could be to remove such data, but it won’t make sense, as it will lead to inaccuracies. 

Q2. What metric would you report for this dataset?

In [38]:
total=0
for i in range(len(data)):
  total+=data["order_amount"][i]/data["total_items"][i]
new_metric=total/len(data)

A good metric would be to evaluate the amount per item and then take an average over the number of orders, i.e. 5000. For example, considering the 1st order in the dataset, the order_amount value is 224, and the total_items are 2, so the amount per item is 224/2=112. We evaluate the same for all orders, take its sum and then divide by 5000 to get this metric value.


Q3. What is its value?

In [39]:
new_metric

387.7428

With the new metric, we get a value of $ 387.7428. This metric is more suitable than AOV, as it gives a more realistic value for a single unit of a sneaker, instead of measuring it per order.